### ISSUE TITLE
Debug queries

### ISSUE THREAD

**Issue by josser** [neo4j/neo4j-javascript-driver#411]
Hi! it would be great to have ability to specify some option in config to see what actual was sent to server for debugging purposes. 

It could be just simple: 'debug': true / false
There is nice module named debug - https://github.com/visionmedia/debug#readme
Or maybe like it's done in sequelize: you can specify logging function which will be executed for every query sent to server

Thank you!

**Comment by lutovich** [comment]
Hi @josser,

1.7 version of the driver, which is now in a pre-release stage, allows configuring a log level and a logging function. At debug level, this function will receive full Bolt messages as strings. Messages will contain executed queries, returned records, and information about connection pooling. Please see [this PR](https://github.com/neo4j/neo4j-javascript-driver/pull/380) for more details.

It would be really great if you could try out 1.7 pre-release version with logging and give feedback! It can be installed with `npm install --save neo4j-driver@next` and driver with debug logging to console can be created like this:

```javascript
const config = {
  logging: neo4j.logging.console('debug')
};

const driver = neo4j.driver(
 'bolt://localhost',
  neo4j.auth.basic('neo4j', 'password'),
  config
);
```

Thanks in advance!

**Comment by purplemana** [comment]
My project already has a Winston instance, can I pass that to the new driver?

**Comment by lutovich** [comment]
@purplemana `logging` property in the config is an object which allows you to supply the desired log level and a function to do the logging. Function `neo4j.logging.console` is just a convenience factory for such objects. It is thus possible to redirect driver logging to any destination of framework. Here is a link to the API doc: [link](https://github.com/neo4j/neo4j-javascript-driver/blob/1.7.0-beta02/src/v1/index.js#L190-L204). It is not yet in the public API doc because 1.7 branch is in a pre-release state.

`logging` config property has the following shape:

```
{
  level: 'error' | 'warn' | 'info' | 'debug',
  logger: function(level, message) {...}
}
```

For Winston, implementation could be smth like:

```javascript
var winstonLogger = createLogger(...);

var config = {
  logging: {
    level: 'info',
    logger: function(level, message) { winstonLogger.log(level, message); }
  }
};

var driver = neo4j.driver('bolt://localhost', neo4j.auth.basic('neo4j', 'password'), config);
```

It would be great if you could try it out and give feedback!

**Comment by lutovich** [comment]
Hey @josser, @purplemana!

Did you have a chance to try out new logging?

**Comment by josser** [comment]
Oh, sorry I didn't have a chance to try it, it was small "investigation" project, but I'll definitely try this in new projects because we have plans to use neo4j in future

**Comment by CharlBest** [comment]
I will try it very soon.

**Comment by lutovich** [comment]
Mentioned logging functionality is now available in 1.7.0 version of the driver.

**Comment by CharlBest** [comment]
This works really nicely. Thanks for the work!
